crypto/fipsmodule/aes/aes_test.cc

```diff
@@ -479,6 +479,17 @@ TEST(AESTest, ABI) {
         SCOPED_TRACE(blocks);
         CHECK_ABI(aes_hw_cbc_encrypt, buf, buf, AES_BLOCK_SIZE * blocks, &key,
                   block, AES_ENCRYPT);
+        if (blocks == 0) {
+          // Without this initialization, valgrind complains
+          // about using an unitialized value.
+          for (size_t i = 0; i < 64; i++) {
+            buf[i] = i;
+          }
+          Bytes buf_before = Bytes(buf,64);
```

Comment:
Bytes/span doesn't create a copy of the data, they reference the original memory. So this test passes even if aes_hw_encrypt touches buf. For example this passes:
```
TEST(BytesTest, copyOrClone) {
  uint8_t buf[64] = {0};
  Bytes buf_before = Bytes(buf,64);
  buf[0] = 1;
  ASSERT_EQ(buf_before, Bytes(buf, 64));
}
```

Reply:
Thanks, Andrew. I brought back the string comparison I had as I was testing in https://github.com/aws/aws-lc/pull/1690/commits/110ab16bdea385c42881473e71f677043b75ae59.